### ISSUE TITLE
Ownership handover

### ISSUE THREAD

**Issue by jussi-kalliokoski** [jussi-kalliokoski/audiolib.js#82]
Hello everyone,

First of all, I'd like to thank you for the great support everyone has given me with this project. I've had a blast and it's been a great learning experience. I also hope that during the past few years this project has been of use to people.

However, it has probably become clear to everyone involved that I am no longer unable to maintain this library. I haven't made significant commits to the project since last year, aside from a few commits I just made to fix compatibility with node.js 0.10.x and Grunt 0.4.x.

I may be able to fix a few bugs here and there, but implementing features will no longer happen. Let me elaborate...

The whole project is currently a huge mess. At the beginning of this project, it was just me messing around, trying to understand some concepts of DSP. The library started out as just a JS oscillator implementation. As a result of those things and my general inexperience, I made some terrible architectural choices that bit back quite fast. To name a few: single-sample-based processing, non-modularity and inheritance model. Those problems just got worse by time as the need for this sort of library grew immensely and the library grew from an oscillator implementation to a do-it-all audio framework.

At one point, I managed to fix the single most horrible thing from the implementation perspective: the inheritance model. Using prototype-based inheritance enabled JS engines to optimize to internal classes and the perfomance boost was immense, not to mention a much cleaner implementation. However, as a side effect, I did some stupid stuff to make the inheritance and creating instances "simpler", in one way polluting the codebase even further.

After that I started the effort of trying to make the code base more modular by stripping off parts of audiolib.js (e.g. sink.js) as their own libraries. For some reason, however, I decided to maintain backwards compatibility and keep the parts also included in audiolib.js so that people would not have to have a dozen of different libraries to get the job done. This ended up being a maintenance nightmare.

As it is currently, I no longer think audiolib.js serves the purpose it was designed for very well. In fact, I'd recommend against using it. For a lot of its use cases, it can be substituted with Web Audio API.

I'm however working on, when I can, a complete "rewrite" of the library in the `next` branch of the repo. This will feature a completely different API and probably will even be released under a different name. My design goals are minimalism, extensibility, modularity and performance. The new library will take advantage of existing work if available, such as Web Audio API, rather than try to work around them or abstract them away.

Long story short, I will no longer be the maintainer of this project. If someone wants to claim ownership and either maintain the current version or slowly iterate on it, I'd be glad to hand over the ownership.

Thanks everyone, and sorry. It's been a great ride. I hope we can make more great audio libraries together in the future!
